src/ORM/Association/ExternalAssociationTrait.php

```diff
@@ -57,7 +57,7 @@ public function foreignKey($key = null)
     {
         if ($key === null) {
             if ($this->_foreignKey === null) {
-                $this->_foreignKey = $this->_modelKey($this->source()->alias());
+                $this->_foreignKey = $this->_modelKey($this->source()->table());
```

Comment:
This is the only way I found to correct this bug. I'm afraid finding something else is just beyond my skills :(